File: 49-group-anagrams/49-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> m;
        for(int i=0;i<strs.size();i++){
            string sorted_s = strs[i];
            sort(sorted_s.begin(), sorted_s.end());
            if(m.find(sorted_s) != m.end()){
                (m.find(sorted_s)->second).push_back(strs[i]);
            }
            else{
                vector<string> v;
                v.push_back(strs[i]);
                m[sorted_s] = v;
            }
        }
        for(auto it : m){
            res.push_back(it.second);
        }
        return res;
        
    }
};

File: 49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = {}
        for s in strs:
            sorted_s = ''.join(sorted(s))
            if sorted_s in m:
                m[sorted_s].append(s)
            else:
                m[sorted_s] = [s]
        res = []
        for key, val in m.items():
            res.append(val)
        return res
        